homeassistant/components/openweathermap/weather_update_coordinator.py

```
@@ -267,7 +267,7 @@ def _get_condition(self, weather_code, timestamp=None):
                 return ATTR_CONDITION_SUNNY
             return ATTR_CONDITION_CLEAR_NIGHT
 
-        return [k for k, v in CONDITION_CLASSES.items() if weather_code in v][0]
+        return CONDITION_MAP.get(weather_code)
```

This would previously fail hard